File: frontend/src/services/authService.js
// Mạnh Hùng - HE200743
// Service xử lý các thao tác xác thực: đăng nhập, đăng ký tài khoản và đổi mật khẩu.
// Tất cả các hàm đều gọi API backend thông qua axiosClient đã cấu hình sẵn base URL và token.
import axiosClient from '../api/axiosClient'

const authService = {
  // Đăng nhập bằng email và mật khẩu, trả về token JWT cùng thông tin người dùng
  login: (credentials) => axiosClient.post('/v1/auth/login', credentials),

  // Đăng nhập bằng tài khoản Google: gửi ID token lên backend để xác minh và tạo/đăng nhập tài khoản
  loginWithGoogle: (idToken) => axiosClient.post('/v1/auth/google', { idToken }),

  // Đăng ký tài khoản bệnh nhân mới với họ tên, email, số điện thoại và mật khẩu
  register: (data) => axiosClient.post('/v1/auth/register', data),

  // Đổi mật khẩu cho người dùng đang đăng nhập (yêu cầu mật khẩu hiện tại và mật khẩu mới)
  changePassword: (data) => axiosClient.put('/v1/auth/change-password', data),
}

export default authService

File: backend/src/main/java/com/ecms/service/AuthService.java
// Mạnh Hùng - HE200743
// Interface định nghĩa các hành vi xác thực người dùng: đăng nhập, đăng ký và đổi mật khẩu.
// Được triển khai bởi AuthServiceImpl.
package com.ecms.service;

import com.ecms.dto.request.ChangePasswordRequest;
import com.ecms.dto.request.GoogleLoginRequest;
import com.ecms.dto.request.LoginRequest;
import com.ecms.dto.request.RegisterRequest;
import com.ecms.dto.response.AuthResponse;

public interface AuthService {
    // Xác thực thông tin đăng nhập và trả về token JWT cùng thông tin người dùng
    AuthResponse login(LoginRequest request);

    // Tạo tài khoản bệnh nhân mới và trả về token để tự động đăng nhập
    AuthResponse register(RegisterRequest request);

    // Đăng nhập/đăng ký bằng tài khoản Google: xác minh ID token, tìm hoặc tạo tài khoản với vai trò PATIENT
    AuthResponse loginWithGoogle(GoogleLoginRequest request);

    // Thay đổi mật khẩu cho người dùng có email tương ứng sau khi xác minh mật khẩu hiện tại
    void changePassword(String email, ChangePasswordRequest request);
}
